### PR TITLE
[CONTINT-4069] Fix flaky sbom queue tests

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR fixes a flaky test.

```
WARNING: DATA RACE
Write at 0x00c0003feff0 by goroutine 138:
  runtime.mapassign_faststr()
      /usr/local/go/src/runtime/map_faststr.go:203 +0x0
  github.com/DataDog/datadog-agent/pkg/sbom/collectors.RegisterCollector()
      /go/src/github.com/DataDog/datadog-agent/pkg/sbom/collectors/collectors.go:57 +0x4db
  github.com/DataDog/datadog-agent/pkg/sbom/scanner.TestRetryLogic_Host()
      /go/src/github.com/DataDog/datadog-agent/pkg/sbom/scanner/scanner_test.go:215 +0x4b0
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1595 +0x261
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1648 +0x44
Previous read at 0x00c0003feff0 by goroutine 132:
  runtime.mapiternext()
      /usr/local/go/src/runtime/map.go:867 +0x0
  github.com/DataDog/datadog-agent/pkg/sbom/scanner.(*Scanner).startScanRequestHandler.func2()
```
This stack trace indicates that a test is updating the `Collectors` global map while another goroutine is closing every collectors. Collectors are closed only when the main goroutine is canceled, which occurs when a test terminates and we cancel the context. It means that while `test 2` starts, `test 1` hasn't terminated yet.

We can't just wait that the first goroutine terminates so I propose to remove part of the global state.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

File: pkg/sbom/scanner/scanner_test.go [hunk 1]
```diff
@@ -199,6 +199,8 @@ func TestRetryLogic_ImageDeleted(t *testing.T) {
 		}
 	}, 15*time.Second, 1*time.Second)
 	cancel()
+	// Ensure the collector is stopped
+	shutdown.WaitUntil(time.After(2 * time.Second))
```
Comment (clamoriniere) [inline pkg/sbom/scanner/scanner_test.go:203]: Can we get the collectors status?
It would be nice if collectors have a IsStoped() function

**Review by clamoriniere — COMMENTED** [review]
It would be nice to avoid relying on a Sleep() function to avoid flakiness

**Pull request merged**